server/src/routes/donatedItemStatusRoutes.ts

```diff
@@ -2,15 +2,46 @@ import { Router, Request, Response } from 'express';
 import prisma from '../prismaClient'; // Import Prisma client
 import { donatedItemStatusValidator } from '../validators/donatedItemStatusValidator'; // Import the validator
 
+import multer from 'multer';
+import {
+    uploadToStorage,
+    getFileExtension,
+} from '../services/donatedItemService';
+
 const router = Router();
 
+const upload = multer({ storage: multer.memoryStorage() });
+
 // PUT /donatedItem/status/:id - Update the status of a DonatedItem
 router.put(
     '/:id',
+    [upload.array('imageFiles'), donatedItemStatusValidator],
     donatedItemStatusValidator,
     async (req: Request, res: Response) => {
         try {
+            const donatedItemId = Number(req.params.id);
+
             const { statusType, dateModified } = req.body;
+            const imageFiles = req.files as Express.Multer.File[];
+
+            if (!statusType) {
+                return res.status(400).json({ error: 'status is required' });
+            }
+            const imageUrls = await Promise.all(
+                imageFiles.map(async file => {
+                    const fileExtension = getFileExtension(file.mimetype);
+                    const uniqueFilename = `status-${donatedItemId}-${Date.now()}${fileExtension}`;
+                    return await uploadToStorage(file, uniqueFilename);
+                }),
+            );
+
+            const updatedItem = await prisma.donatedItem.update({
+                where: { id: donatedItemId },
+                data: {
+                    currentStatus: statusType,
+                    lastUpdated: new Date(),
+                },
```

Comment:
Already prisma create and update methods are in place [as shown ](https://github.com/oss-slu/material-donor-mutual-assist/blob/main/server/src/routes/donatedItemStatusRoutes.ts#L16C12-L33C16). Just adding `imageUrls `to .update method would work.